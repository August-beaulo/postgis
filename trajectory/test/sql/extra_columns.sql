--
-- check_postgis_installed
--

\c trjregression

SET client_min_messages TO warning;

-- step 0: add a trajectory
SELECT trajectory.CreateTrajectory('taxi', 'B130',  4326, 0.00001);

-- step 1: Add a column
SELECT trajectory.AddTrajectoryColumns('taxi', 'speed real DEFAULT 0.0');
SELECT trajectory.AddTrajectoryColumns('taxi', 'direction real DEFAULT 0.0 CHECK (direction <= 360)');

-- query it now
\d trajectory.taxi

-- step 2: Append with old function
SELECT trajectory.AppendTrajectory('taxi', 'B130', TIMESTAMP '2015-10-10 8:10:00', ST_GeomFromText('POINT(119.4 39.4)', 4326));

-- step 3: Append with new function
SELECT trajectory.AppendTrajectoryWithAttr('taxi', 'B130', TIMESTAMP '2015-10-10 8:15:00', ST_GeomFromText('POINT(119.4 39.4)', 4326), '1.1' );
SELECT trajectory.AppendTrajectoryWithAttr('taxi', 'B130', TIMESTAMP '2015-10-10 8:20:00', ST_GeomFromText('POINT(119.4 39.4)', 4326), '2.2', '20' );
SELECT trajectory.AppendTrajectoryWithAttr('taxi', 'B130', TIMESTAMP '2015-10-10 8:25:00', ST_GeomFromText('POINT(119.4 39.4)', 4326), '3.3', '30', '30' );

-- query it now
SELECT id,time,speed,direction FROM trajectory.taxi ORDER BY id;

-- step 4: Drop a column
SELECT trajectory.DropTrajectoryColumns('taxi', 'speed');

-- query it now
\d trajectory.taxi

-- step 5: Append with new function
SELECT trajectory.AppendTrajectoryWithAttr('taxi', 'B130', TIMESTAMP '2015-10-10 8:30:00',, ST_GeomFromText('POINT(119.4 39.4)', 4326), '40' );

-- step 6: error test
SELECT trajectory.AppendTrajectoryWithAttr('taxi', 'B130', TIMESTAMP '2015-10-10 8:35:00',, ST_GeomFromText('POINT(119.4 39.4)', 4326), '500' );

-- query it now
SELECT id,time,direction FROM trajectory.taxi ORDER BY id;

-- drop it
SELECT trajectory.DeleteTrajectory('taxi', 'B130');
