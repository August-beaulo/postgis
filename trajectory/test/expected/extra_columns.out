--
-- check_postgis_installed
--
\c trjregression
SET client_min_messages TO warning;
-- step 0: add a trajectory
SELECT trajectory.CreateTrajectory('taxi', 'B130',  4326, 0.00001);
 createtrajectory 
------------------
               13
(1 row)

-- step 1: Add a column
SELECT trajectory.AddTrajectoryColumns('taxi', 'speed real DEFAULT 0.0');
     addtrajectorycolumns     
------------------------------
 trajectory.taxi.speed added.
(1 row)

SELECT trajectory.AddTrajectoryColumns('taxi', 'direction real DEFAULT 0.0 CHECK (direction <= 360)');
       addtrajectorycolumns       
----------------------------------
 trajectory.taxi.direction added.
(1 row)

-- query it now
\d trajectory.taxi
                             Append-Only Table "trajectory.taxi"
  Column   |            Type             |                     Modifiers                     
-----------+-----------------------------+---------------------------------------------------
 id        | bigint                      | not null default nextval('taxi_id_seq'::regclass)
 time      | timestamp without time zone | not null
 position  | geometry(Point,4326)        | not null
 speed     | real                        | default 0.0
 direction | real                        | default 0.0
Compression Type: None
Compression Level: 0
Block Size: 32768
Checksum: t
Indexes:
    "pool_taxi_position_gist" gist ("position")
    "pool_taxi_time_btree" btree ("time")
Check constraints:
    "taxi_direction_check" CHECK (direction <= 360::double precision)
Distributed by: (id)

-- step 2: Append with old function
SELECT trajectory.AppendTrajectory('taxi', 'B130', TIMESTAMP '2015-10-10 8:10:00', ST_GeomFromText('POINT(119.4 39.4)', 4326));
 appendtrajectory 
------------------
 t
(1 row)

-- step 3: Append with new function
SELECT trajectory.AppendTrajectoryWithAttr('taxi', 'B130', TIMESTAMP '2015-10-10 8:15:00', ST_GeomFromText('POINT(119.4 39.4)', 4326), '1.1' );
WARNING:  trajectory.AppendTrajectoryWithAttr() - Input attributes #1 != attribute columns #2.
 appendtrajectorywithattr 
--------------------------
 t
(1 row)

SELECT trajectory.AppendTrajectoryWithAttr('taxi', 'B130', TIMESTAMP '2015-10-10 8:20:00', ST_GeomFromText('POINT(119.4 39.4)', 4326), '2.2', '20' );
 appendtrajectorywithattr 
--------------------------
 t
(1 row)

SELECT trajectory.AppendTrajectoryWithAttr('taxi', 'B130', TIMESTAMP '2015-10-10 8:25:00', ST_GeomFromText('POINT(119.4 39.4)', 4326), '3.3', '30', '30' );
WARNING:  trajectory.AppendTrajectoryWithAttr() - Input attributes #3 != attribute columns #2.
ERROR:  INSERT has more expressions than target columns
CONTEXT:  SQL statement "INSERT INTO trajectory.taxi VALUES (13,'Sat Oct 10 08:25:00 2015','0101000020E61000009A99999999D95D403333333333B34340','3.3','30','30')"
PL/pgSQL function "appendtrajectorywithattr" line 58 at execute statement
-- query it now
SELECT id,time,speed,direction FROM trajectory.taxi ORDER BY id;
 id |           time           | speed | direction 
----+--------------------------+-------+-----------
 13 | Sat Oct 10 08:15:00 2015 |   1.1 |         0
 13 | Sat Oct 10 08:20:00 2015 |   2.2 |        20
 13 | Sat Oct 10 08:10:00 2015 |     0 |         0
(3 rows)

-- step 4: Drop a column
SELECT trajectory.DropTrajectoryColumns('taxi', 'speed');
     droptrajectorycolumns      
--------------------------------
 trajectory.taxi.speed removed.
(1 row)

-- query it now
\d trajectory.taxi
                             Append-Only Table "trajectory.taxi"
  Column   |            Type             |                     Modifiers                     
-----------+-----------------------------+---------------------------------------------------
 id        | bigint                      | not null default nextval('taxi_id_seq'::regclass)
 time      | timestamp without time zone | not null
 position  | geometry(Point,4326)        | not null
 direction | real                        | default 0.0
Compression Type: None
Compression Level: 0
Block Size: 32768
Checksum: t
Indexes:
    "pool_taxi_position_gist" gist ("position")
    "pool_taxi_time_btree" btree ("time")
Check constraints:
    "taxi_direction_check" CHECK (direction <= 360::double precision)
Distributed by: (id)

-- step 5: Append with new function
SELECT trajectory.AppendTrajectoryWithAttr('taxi', 'B130', TIMESTAMP '2015-10-10 8:30:00',, ST_GeomFromText('POINT(119.4 39.4)', 4326), '40' );
ERROR:  syntax error at or near ","
LINE 1: ...tr('taxi', 'B130', TIMESTAMP '2015-10-10 8:30:00',, ST_GeomF...
                                                             ^
-- step 6: error test
SELECT trajectory.AppendTrajectoryWithAttr('taxi', 'B130', TIMESTAMP '2015-10-10 8:35:00',, ST_GeomFromText('POINT(119.4 39.4)', 4326), '500' );
ERROR:  syntax error at or near ","
LINE 1: ...tr('taxi', 'B130', TIMESTAMP '2015-10-10 8:35:00',, ST_GeomF...
                                                             ^
-- query it now
SELECT id,time,direction FROM trajectory.taxi ORDER BY id;
 id |           time           | direction 
----+--------------------------+-----------
 13 | Sat Oct 10 08:15:00 2015 |         0
 13 | Sat Oct 10 08:20:00 2015 |        20
 13 | Sat Oct 10 08:10:00 2015 |         0
(3 rows)

-- drop it
SELECT trajectory.DeleteTrajectory('taxi', 'B130');
                        deletetrajectory                        
----------------------------------------------------------------
 Trajectory taxi (B130) (with id = 13) is removed successfully.
(1 row)

